robot/impl/resource_manager.go

```diff
@@ -830,20 +838,47 @@ func (manager *resourceManager) updateResources(
 	defer manager.configLock.Unlock()
 	var allErrs error
 
+	// stores resource names so that if there is duplicate we error
```

Comment:
I'm suspicious of storing all types of resources in the same seen map. Can we definitely confirm that resources across types can have the same "name" but different stringified resources names? What about a process and a module with the same name?

Also, to be clear, the resource graph (note that modules and processes are not in the resource graph), cannot actually contain duplicate components or services since it's a `map[resource.Name]GraphNode`. I assume you just want to protect users against silently failing though.

Reply:
Oh sorry just saw the existing convo on this; seems like you all are aware of this already.

Reply:
yup! this is to protect against silent failures. Do you see any issues with putting it all in the same map?